### DISCUSSION TITLE
404 error when open tool page

### DISCUSSION THREAD

**Discussion by s-fog** [laravel/nova-issues#5093] [Q&A]
- Laravel Version: 9.40.1
- Nova Version: 4.19.0
- PHP Version: 8.1.1
- Database Driver & Version: MYSQL

### Description:

Iam create tool and develop it with success. Url in localhost is http://localhost:16275/manager-tool.

Then iam deploy this to staging server and when i open it, i have 404 error. https://domain.domain/manager-tool

What iam doing wrong?

I push all code to remote server and launch this deploy process

docker exec dog_php_staging composer install
docker exec dog_php_staging php artisan config:cache
docker exec dog_php_staging php artisan route:cache
docker exec dog_php_staging php artisan view:cache
docker exec dog_php_staging php artisan migrate --force

**Comment by crynobone** [comment]
You can start debugging the issue using 

```php
php artisan route:list
```

**Reply by s-fog** [reply]
Thanks for your answer !

Yes, i have this route on localhost and server

Localhost
![image](https://user-images.githubusercontent.com/10043516/202937150-aec40e7b-4662-41db-84a7-42c74b62ac36.png)

Server
![image](https://user-images.githubusercontent.com/10043516/202937213-80660a51-99b9-4343-a612-f839df5e93bd.png)

Other routes that i defined in api.php file also here

**Comment by crynobone** [comment]
It seems the route locally is `admin/manager-tool` and on the production `/manager-tool`. You need to debug all the route/link are handled properly.

**Reply by s-fog** [reply]
It is not the case, because every other routes, builded from nova are working. 

Localhost - http://localhost:16275/admin/resources/contact-nova-resources
Server - https://admin.test.dog-ville.ru/resources/contact-nova-resources

Config nova.php
```
'domain' => env('NOVA_DOMAIN_NAME', null),
    'nova_url' => env('NOVA_URL', 'http://localhost'),
    'url' => env('APP_URL', '/'),
    'path' => env('NOVA_PATH', '/'),
```

nova_url - is my custom variable, url - i think too

localhost .env
```
APP_URL=http://localhost:16275
NOVA_URL=http://localhost:16275/admin
NOVA_PATH=/admin
```

server .env
```
APP_URL=https://test.dog-ville.ru
NOVA_DOMAIN_NAME=admin.test.dog-ville.ru
NOVA_URL=https://admin.test.dog-ville.ru
```

**Reply by s-fog** [reply]
Any help?

**Reply by crynobone** [reply]
Sorry but I can't help on an issue I can't reproduced

**Reply by s-fog** [reply]
Great. Why i paid for product which doesn't work ?

**Reply by s-fog** [reply]
Get here guy that can help with issue?

**Reply by s-fog** [reply]
I have my hole project on this admin panel, will be very hard to move somewhere else. Need help

**Reply by crynobone** [reply]
> Great. Why i paid for product which doesn't work ?

The code is working fine and the above issue isn't reproducible in "normal" Nova installation: 

https://user-images.githubusercontent.com/172966/203015970-f11d12f4-817c-41ec-8175-ec7c9dab5c24.mp4

I don't deny that you are having issue, but you can't expect anyone can help with just minimum information.

**Reply by s-fog** [reply]
Ask me more. I dont know what exactly can help

**Reply by crynobone** [reply]
![image](https://user-images.githubusercontent.com/172966/203019063-ea8b1723-0b06-4f01-8957-a9d52b004276.png)

Tested with NGROK with the same configuration, also Nova working without any bugs.

**Reply by s-fog** [reply]
iI know that everything needs to be working, but it is not, can i share project to u ?

**Reply by s-fog** [reply]
> ![image](https://user-images.githubusercontent.com/172966/203019063-ea8b1723-0b06-4f01-8957-a9d52b004276.png)
> 
> Tested with NGROK with the same configuration, also Nova working without any bugs.

Show me .env file

**Reply by crynobone** [reply]
https://github.com/nova-issues/issue-5093/commit/2e202a95604b8a4529d354f5212b88ae72d7a77e

**Reply by s-fog** [reply]
[ManagerTool.zip](https://github.com/laravel/nova-issues/files/10055582/ManagerTool.zip)

Can u try this?

**Reply by s-fog** [reply]
Resume, when 'path' => env('NOVA_PATH', '/'), then https://admin.test.dog-ville.ru/manager-tool 404.
when 'path' => env('NOVA_PATH', '/admin'), then https://admin.test.dog-ville.ru/admin/manager-tool everything works fine, but this url opening main website https://admin.test.dog-ville.ru/
I want nova be on separate domain

admin.test.dog-ville.ru for nova
test.dog-ville.ru for website

**Reply by s-fog** [reply]
'path' => env('NOVA_PATH', '/admin') In this configuration all admin panel works as expected, but tool is not

**Reply by s-fog** [reply]
Can we fix this case?

**Reply by crynobone** [reply]
Unable to replicate the issue. 

Laravel app: http://issue-5093.test
Nova app: http://nova.issue-5093.test

```ini
APP_URL=http://issue-5093.test
NOVA_DOMAIN_NAME=nova.issue-5093.test
```

https://user-images.githubusercontent.com/172966/203183873-2381ae6b-c7da-4d6f-affc-b07625c820ac.mp4

![image](https://user-images.githubusercontent.com/172966/203187326-4dc73ec4-377e-4f46-94a0-c97880e8f787.png)

We no longer looking into this.

1. I believe this is a usage issue, and you need to fix it.
2. We already have multiple feature tests covering the use case 
3. I already demonstrate there no issue using Nova as separate domain
